File: python/Level 1/최대공약수와_최소공배수.py
def gcd(a, b):
    while True:
        if a > b:
            large, small = a, b
        else:
            large, small = b, a

        if large % small == 0:
            return small

        a = large % small
        b = small


def solution(n, m):
    d = gcd(n, m)
    return [d, n*m // d]

File: golang/Level 1/최대공약수와_최소공배수.go
func gcd(a int, b int) int {
    var large, small int
    
    for {
        if a > b {
            large = a
            small = b
        } else {
            large = b
            small = a
        }
        
        if large % small == 0 {
            return small
        }
        
        a = large % small
        b = small
    }
}


func solution(n int, m int) []int {
    d := gcd(n, m)
    return []int{d, n*m / d}
}
